GUI/Main.cs

```diff
@@ -495,42 +498,82 @@ private void OnFilterUpdateTimer(Object source, EventArgs e)
         private void ModList_KeyPress(object sender, KeyPressEventArgs e)
         {
+            var current_row = ModList.CurrentRow;
+            var key = e.KeyChar.ToString();
+
             // Check the key. If it is space, mark the current mod as selected.
-            if (e.KeyChar.ToString() == " ")
+            if (key == " ")
             {
-                var selected_row = ModList.CurrentRow;
-
-                if (selected_row != null)
+                if (current_row != null && current_row.Selected)
                 {
                     // Get the checkbox.
-                    var selected_row_check_box = selected_row.Cells["Installed"] as DataGridViewCheckBoxCell;
+                    var selected_row_check_box = current_row.Cells["Installed"] as DataGridViewCheckBoxCell;
 
                     // Invert the value.
                     if (selected_row_check_box != null)
                     {
                         bool selected_value = (bool)selected_row_check_box.Value;
                         selected_row_check_box.Value = !selected_value;
-                    }                    
+                    }
                 }
                 e.Handled = true;
                 return;
             }
 
             var rows = ModList.Rows.Cast<DataGridViewRow>().Where(row => row.Visible);
-            var does_name_begin_with_char = new Func<DataGridViewRow, bool>(row =>
+
+            // Determine time passed since last key press
+            TimeSpan interval = DateTime.Now - lastSearchTime;
+            if (interval.TotalSeconds < 1) {
+                // Last keypress was < 1 sec ago, so combine the last and current keys
+                key = lastSearchKey + key;
+            }
+            // Remember the current time and key
+            lastSearchTime = DateTime.Now;
+            lastSearchKey = key;
+
+            if (key.Distinct().Count() == 1)
+            {
+                // It's the same key being pressed repeatedly, so use only that
+                key = key.Substring(0, 1);
+            }
+
+            var current_name = ((GUIMod) current_row.Tag).ToCkanModule().name;
+            var current_match = current_name.StartsWith(key, StringComparison.OrdinalIgnoreCase);
+            DataGridViewRow first_match = null;
+
+            var does_name_begin_with_key = new Func<DataGridViewRow, bool>(row =>
             {
                 var modname = ((GUIMod) row.Tag).ToCkanModule().name;
-                var key = e.KeyChar.ToString();
-                return modname.StartsWith(key, StringComparison.OrdinalIgnoreCase);
+                var row_match = modname.StartsWith(key, StringComparison.OrdinalIgnoreCase);
+                if (row_match && first_match == null) {
+                    // Remember the first match to allow cycling back to it if necessary
+                    first_match = row;
+                }
+                if (row == current_row || (current_match && row.Index < current_row.Index))
```

Comment:
Why have this in? If I type better I would expect, after bet, to have Better Buoyancy selected and for it to remain on that row instead of cycling through the next couple which also starts with better.